src/views/hostnetworkconfiguration/components/TopologySidebar/TopologySidebar.tsx

```diff
@@ -15,26 +21,52 @@ type TopologySidebarProps = {
   setSelectedIds: Dispatch<SetStateAction<string[]>>;
 };
 const TopologySidebar: FC<TopologySidebarProps> = ({ states, selectedIds, setSelectedIds }) => {
-  const { selectedState, selectedInterface } = useMemo(() => {
-    if (selectedIds.length === 0) return { selectedState: null, selectedInterface: null };
+  const [policies] = useK8sWatchResource<V1beta1NodeNetworkConfigurationPolicy[]>({
```

Comment:
Consider moving all of this logic to a hook to separate it from the presentation.